rsocket-tck-drivers/src/test/java/io/rsocket/tckdrivers/TckTest.java

```diff
@@ -79,19 +87,50 @@ public void TckTestRunner() {
 
     assertNotNull("JavaClientDriver is not defined", jd);
     try {
-      jd.runTest(this.test.subList(1, this.test.size()), this.name);
+      jd.runTest(this.tckTest.test.subList(1, this.tckTest.test.size()), this.tckTest.name);
     } catch (Exception e) {
       e.printStackTrace();
     }
   }
 
   /**
+   * A function that parses the file and extract the individual tests
+   *
+   * @param file: file to read as input.
+   * @return a list of individual tests. Each individual test is also a list of String.
+   */
+  private static List<List<String>> ExtractTests(File file) throws Exception {
+
+    BufferedReader reader = new BufferedReader(new FileReader(file));
```

Comment:
Suggestion: why not try using Reactor library here?  As a *optional* programming exercise, what does this method look like if you refactor it to use methods that only use reactor types as the reponses e.g. Flux<TckIndividualTest>, Flux<File>, Flux<String>